_books/ion-1-1/src/modules/defining_modules.md

```diff
@@ -0,0 +1,242 @@
+)
+```
+
+
+
+### `macro_table`
+
+Macros are declared after symbols.
+The `macro_table` clause assembles a list of macro definitions for the module to export. It takes any number of arguments.
+All modules have a macro table, so when a module has no `macro_table` clause, the module has an empty macro table.
+
+Most commonly, a macro table entry is a definition of a new macro expansion function, following
+this general shape:
+
+When no name is given, this defines an anonymous macro that can be referenced by its numeric
+address (that is, its index in the enclosing macro table).
+Inside the defining module, that uses a local reference like `12`.
+
+The _signature_ defines the syntactic shape of expressions invoking the macro;
+see [Macro Signatures](../macros/defining_macros.md#macro-signatures) for details.
+The _template_ defines the expansion of the macro, in terms of the signature’s parameters;
+see [Template Expressions](../macros/defining_macros.md#template-definition-language-tdl) for details.
+
+Imported macros must be explicitly exported if so desired.
+Module names and `export` clauses can be intermingled with `macro` definitions inside the `macro_table`;
+together, they determine the bindings that make up the module’s exported macro array.
+
+The _module-name_ export form is shorthand for referencing all exported macros from that module,
+in their original order with their original names.
+
+An `export` clause contains a single macro reference followed by an optional alias for the exported macro.
+The referenced macro is appended to the macro table.
+
+> [!TIP]
+> No name can be repeated among the exported macros, including macro definitions.
+> Name conflicts must be resolved by `export`s with aliases.
+
+#### Processing
+
+When the `macro_table` clause is encountered, the reader constructs an empty list. The arguments to the clause are then processed from left to right.
+
+For each `arg`:
+* **If the `arg` is a `macro` clause**, the clause is processed and the resulting macro definition is appended 
+  to the end of the macro table being constructed.
+* **If the `arg` is an `export` clause**, the clause is processed and the referenced macro definition is appended 
+  to the end of the macro table being constructed.
+* **If the `arg` is the name of a module**, the macro definitions in that module's macro table are appended
+  to the end of the macro table being constructed.
+* **If the `arg` is anything else**, the reader must raise an error.
+
+
+A macro name is a symbol that can be used to reference a macro, both inside and outside the module.
+Macro names are optional, and improve legibility when using, writing, and debugging macros.
+When a name is used, it must be an identifier per Ion’s syntax for symbols.
+Macro definitions being added to the macro table must have a unique name.
+If a macro is added whose name conflicts with one already present in the table, the implementation must raise an error.
```

Comment:
Except for the system macros, which may be shadowed?

Reply:
They can only be shadowed in the sense that a system macro `make_string` can exist in the system macro table at the same time as a user macro `make_string` can exist in any other macro table. However, it is never legal to have two macros given the same name in the same macro table, even if one is a system macro.